src/player.cpp

```diff
@@ -0,0 +1,72 @@
+#include "player.h"
+
+Player::Player() : Entity() {
+	this->setFrameDelay(PLAYER_ANIMATION_DELAY);
+	this->setLoop(false);
+}
+
+Player::~Player() {}
+
+void Player::update(float frameTime, Stage *stage, Input *input, 
+	std::map<std::string, bool> *keysPressed) {
+
+	if (input->isKeyDown(LEFT_KEY) && !(*keysPressed)["LEFT"]) {
+		this->rotateEntity("LEFT", stage->isValidMove(LEFT));
+	}
+
+	if (input->isKeyDown(RIGHT_KEY) && !(*keysPressed)["RIGHT"]) {
+		this->rotateEntity("RIGHT", stage->isValidMove(RIGHT));
+	}
+
+	if (input->isKeyDown(UP_KEY) && !(*keysPressed)["UP"]) {
+		this->rotateEntity("UP", stage->isValidMove(UP));
+	}
+
+	if (input->isKeyDown(DOWN_KEY) && !(*keysPressed)["DOWN"]) {
+		this->rotateEntity("DOWN", stage->isValidMove(DOWN));
+	}
+
+	if (this->getAnimationComplete()) {
+		// Clean up
+		this->setFrames(0, 0);
+		this->setCurrentFrame(PLAYER_STANDING_FRAME);
+	}
+
+	Entity::update(frameTime);
+}
+
+void Player::rotateEntity(std::string direction, bool moveValid) {
```

Comment:
I will change the string directions to int directions in the next pr as quite a lot of stuff needs to be changed.